### ISSUE TITLE
Replay for Vulkan trace is broken

### ISSUE THREAD

**Issue by Qining** [google/gapid#1288]
@pmuetschard I just tried the latest master branch. It breaks with following log.
```
W 07:19:16.504 gapic gapis: 
W 07:19:16.504 gapic gapis: Stacktrace code: ljObAzEI6foKQ5OhYFlIoYFI0yD0Ph0AAIXeX2BNRUkrh2BV6gF5liEM3tSqYHdIoPA6QYH2KrZMJ2JIENjO8EqTWA
W 07:19:16.510 gapic LoadData error
W 07:19:16.510 gapic LoadData error
W 07:19:26.508 gapic gapis: Timeout waiting for cleanuppanic: runtime error: invalid memory address or nil pointer dereference [recovered]
W 07:19:26.509 gapic gapis: 	panic: Panic at command 0:eglGetError() → 12288:
W 07:19:26.509 gapic gapis: runtime error: invalid memory address or nil pointer dereference [recovered]
W 07:19:26.510 gapic gapis: 	panic: 
```

**Comment by ben-clayton** [comment]
The full panic stack trace would be helpful here. If you can  also email me repro steps, I can try to take a look tomorrow.

**Comment by Qining** [comment]
To reproduce, just trace with vulkanBloom.apk and then replay. The device I'm using is Pixel C.

Here is the full panic stack:
```
W 07:33:12.338 gapic gapis: 
W 07:33:12.339 gapic gapis: Stacktrace code: ljObAzEI6foKQ5OhYFlIoYFI0yD0Ph0AAIXeX2BNRUkrh2BV6gF5liEM3tSqYHdIoPA6QYH2KrZMJ2JIENjO8EqTWA
W 07:33:12.345 gapic LoadData error
W 07:33:12.345 gapic LoadData error
W 07:33:22.342 gapic gapis: Timeout waiting for cleanuppanic: runtime error: invalid memory address or nil pointer dereference [recovered]
W 07:33:22.343 gapic gapis: 	panic: Panic at command 788:RecreateState():
W 07:33:22.343 gapic gapis: runtime error: invalid memory address or nil pointer dereference [recovered]
W 07:33:22.343 gapic gapis: 	panic: 
W 07:33:22.344 gapic gapis: --- *resolve.GetResolvable ---
W 07:33:22.344 gapic gapis:  path:<events:<capture:<id:<data:"\247\023S{\270\341:\270\2101\304\025\356T\323\343\264\177\006\244" > > last_in_frame:true > > 
W 07:33:22.344 gapic gapis:  Store():
W 07:33:22.345 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/database/database.go:51  github.com/google/gapid/gapis/database.Store
W 07:33:22.345 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/database/database.go:65  github.com/google/gapid/gapis/database.Build
W 07:33:22.345 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
W 07:33:22.346 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/server/server.go:249  github.com/google/gapid/gapis/server.(*server).Get
W 07:33:22.346 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/server/grpc.go:167  github.com/google/gapid/gapis/server.(*grpcServer).Get
W 07:33:22.346 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/service/service.pb.go:5465  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
W 07:33:22.347 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/core/app/auth/auth.go:82  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
W 07:33:22.347 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/service/service.pb.go:5467  github.com/google/gapid/gapis/service._Gapid_Get_Handler
W 07:33:22.347 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/third_party/src/google.golang.org/grpc/server.go:719  google.golang.org/grpc.(*Server).processUnaryRPC
W 07:33:22.348 gapic gapis: 
W 07:33:22.348 gapic gapis:  Build() #0:
W 07:33:22.348 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/database/database.go:69  github.com/google/gapid/gapis/database.Build
W 07:33:22.348 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
W 07:33:22.349 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/server/server.go:249  github.com/google/gapid/gapis/server.(*server).Get
W 07:33:22.349 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/server/grpc.go:167  github.com/google/gapid/gapis/server.(*grpcServer).Get
W 07:33:22.349 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/service/service.pb.go:5465  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
W 07:33:22.350 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/core/app/auth/auth.go:82  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
W 07:33:22.350 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/service/service.pb.go:5467  github.com/google/gapid/gapis/service._Gapid_Get_Handler
W 07:33:22.350 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/third_party/src/google.golang.org/grpc/server.go:719  google.golang.org/grpc.(*Server).processUnaryRPC
W 07:33:22.351 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/third_party/src/google.golang.org/grpc/server.go:905  google.golang.org/grpc.(*Server).handleStream
W 07:33:22.351 gapic gapis:  [recovered]
W 07:33:22.351 gapic gapis: 	panic: 
W 07:33:22.351 gapic gapis: --- *resolve.GetResolvable ---
W 07:33:22.352 gapic gapis:  path:<events:<capture:<id:<data:"\247\023S{\270\341:\270\2101\304\025\356T\323\343\264\177\006\244" > > last_in_frame:true > > 
W 07:33:22.352 gapic gapis:  Store():
W 07:33:22.352 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/database/database.go:51  github.com/google/gapid/gapis/database.Store
W 07:33:22.353 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/database/database.go:65  github.com/google/gapid/gapis/database.Build
W 07:33:22.353 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
W 07:33:22.353 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/server/server.go:249  github.com/google/gapid/gapis/server.(*server).Get
W 07:33:22.353 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/server/grpc.go:167  github.com/google/gapid/gapis/server.(*grpcServer).Get
W 07:33:22.354 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/service/service.pb.go:5465  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
W 07:33:22.354 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/core/app/auth/auth.go:82  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
W 07:33:22.354 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/service/service.pb.go:5467  github.com/google/gapid/gapis/service._Gapid_Get_Handler
W 07:33:22.355 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/third_party/src/google.golang.org/grpc/server.go:719  google.golang.org/grpc.(*Server).processUnaryRPC
W 07:33:22.355 gapic LoadData error
W 07:33:22.361 gapic gapis: 
W 07:33:22.362 gapic gapis:  Build() #0:
W 07:33:22.363 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/database/database.go:69  github.com/google/gapid/gapis/database.Build
W 07:33:22.363 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/get.go:27  github.com/google/gapid/gapis/resolve.Get
W 07:33:22.368 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/server/server.go:249  github.com/google/gapid/gapis/server.(*server).Get
W 07:33:22.369 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/server/grpc.go:167  github.com/google/gapid/gapis/server.(*grpcServer).Get
W 07:33:22.369 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/service/service.pb.go:5465  github.com/google/gapid/gapis/service._Gapid_Get_Handler.func1
W 07:33:22.370 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/core/app/auth/auth.go:82  github.com/google/gapid/core/app/auth.ServerInterceptor.func1
W 07:33:22.370 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/service/service.pb.go:5467  github.com/google/gapid/gapis/service._Gapid_Get_Handler
W 07:33:22.370 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/third_party/src/google.golang.org/grpc/server.go:719  google.golang.org/grpc.(*Server).processUnaryRPC
W 07:33:22.371 gapic gapis:   /usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/third_party/src/google.golang.org/grpc/server.go:905  google.golang.org/grpc.(*Server).handleStream
W 07:33:22.371 gapic gapis: 
W 07:33:22.376 gapic gapis: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd9c8a6]
W 07:33:22.377 gapic gapis: 
W 07:33:22.377 gapic gapis: goroutine 1243 [running]:
W 07:33:22.377 gapic gapis: github.com/google/gapid/core/app/crash.Crash.func1()
W 07:33:22.378 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/core/app/crash/crash.go:73 +0x115
W 07:33:22.384 gapic gapis: sync.(*Once).Do(0x4a9f778, 0xc4214f5428)
W 07:33:22.384 gapic gapis: 	/usr/local/google/home/qining/Workspace/tools/go/src/sync/once.go:44 +0xbe
W 07:33:22.384 gapic gapis: github.com/google/gapid/core/app/crash.Crash(0x253bfc0, 0xc4227b7060)
W 07:33:22.385 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/core/app/crash/crash.go:74 +0x9c
W 07:33:22.385 gapic gapis: github.com/google/gapid/core/app/crash.handler()
W 07:33:22.385 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/core/app/crash/crash.go:48 +0x52
W 07:33:22.386 gapic gapis: panic(0x253bfc0, 0xc4227b7060)
W 07:33:22.390 gapic gapis: 	/usr/local/google/home/qining/Workspace/tools/go/src/runtime/panic.go:489 +0x2cf
W 07:33:22.391 gapic gapis: github.com/google/gapid/gapis/database.(*memory).resolvePanicHandler(0xc42035c100, 0x41201a0, 0xc4251dd410)
W 07:33:22.391 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/database/debug.go:106 +0x76e
W 07:33:22.392 gapic gapis: panic(0x255a100, 0xc4227b6c30)
W 07:33:22.392 gapic gapis: 	/usr/local/google/home/qining/Workspace/tools/go/src/runtime/panic.go:489 +0x2cf
W 07:33:22.392 gapic gapis: github.com/google/gapid/gapis/api.ForeachCmd.func1(0xc4214f5b30, 0xc4214f5b58)
W 07:33:22.392 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/api/cmd_foreach.go:39 +0x16c
W 07:33:22.393 gapic gapis: panic(0x2622880, 0x40cde70)
W 07:33:22.393 gapic gapis: 	/usr/local/google/home/qining/Workspace/tools/go/src/runtime/panic.go:489 +0x2cf
W 07:33:22.393 gapic gapis: github.com/google/gapid/gapis/resolve.CommandFilters.All(0xc423da8030, 0x2, 0x2, 0x314, 0x4153500, 0xc4203c2150, 0xc423f76ae0, 0xc42001ae80)
W 07:33:22.394 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/filter.go:36 +0x66
W 07:33:22.395 gapic gapis: github.com/google/gapid/gapis/resolve.(CommandFilters).All-fm(0x314, 0x4153500, 0xc4203c2150, 0xc423f76ae0, 0xc423f76ae0)
W 07:33:22.396 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/events.go:51 +0x6a
W 07:33:22.396 gapic gapis: github.com/google/gapid/gapis/resolve.Events.func1(0x41201a0, 0xc423a1c960, 0x314, 0x4153500, 0xc4203c2150, 0x0, 0x0)
W 07:33:22.396 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/events.go:72 +0x112
W 07:33:22.397 gapic gapis: github.com/google/gapid/gapis/api.ForeachCmd(0x41201a0, 0xc4251dd4a0, 0xc422138000, 0xb00, 0xe00, 0xc4214f5d38, 0x0, 0x0)
W 07:33:22.397 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/api/cmd_foreach.go:46 +0x145
W 07:33:22.397 gapic gapis: github.com/google/gapid/gapis/resolve.Events(0x41201a0, 0xc4251dd4a0, 0xc4251bbb20, 0x4120101, 0x29ef9e0, 0xc4251bbb20)
W 07:33:22.398 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/events.go:167 +0x6c7
W 07:33:22.398 gapic gapis: github.com/google/gapid/gapis/resolve.ResolveInternal(0x41201a0, 0xc4251dd4a0, 0x4120d20, 0xc4251bbb20, 0x25694c0, 0xc42038c0d8, 0x4120d20, 0xc4251bbb20)
W 07:33:22.398 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/resolve.go:287 +0x4a4
W 07:33:22.398 gapic gapis: github.com/google/gapid/gapis/resolve.ResolveService(0x41201a0, 0xc4251dd4a0, 0x4120d20, 0xc4251bbb20, 0xc4251dd4a0, 0x2783460, 0xc42002c001, 0x40ed9a0)
W 07:33:22.399 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/resolve.go:248 +0x4d
W 07:33:22.399 gapic gapis: github.com/google/gapid/gapis/resolve.(*GetResolvable).Resolve(0xc42038c0f0, 0x41201a0, 0xc4251dd4a0, 0x40ed9a0, 0xc42038c0f0, 0x1, 0x40f26e0)
W 07:33:22.399 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/get.go:35 +0xa9
W 07:33:22.400 gapic gapis: github.com/google/gapid/gapis/database.(*record).resolve(0xc423a583c0, 0x41201a0, 0xc4251dd410, 0x41201a0, 0xc4251dd410)
W 07:33:22.400 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/database/memory.go:79 +0x8d
W 07:33:22.400 gapic gapis: github.com/google/gapid/gapis/database.(*memory).resolveLocked.func1()
W 07:33:22.401 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/database/memory.go:156 +0x9f
W 07:33:22.401 gapic gapis: github.com/google/gapid/core/app/crash.Go.func1(0xc4251dd440)
W 07:33:22.401 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/core/app/crash/crash.go:57 +0x43
W 07:33:22.402 gapic gapis: created by github.com/google/gapid/core/app/crash.Go
W 07:33:22.402 gapic gapis: 	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/core/app/crash/crash.go:58 +0x3f
W 07:33:22.403 gapic The gapis process exited with a non-zero exit value: 2

```

**Comment by ben-clayton** [comment]
Awesome, thank you. I'll take a look tomorrow.

**Comment by dsrbecky** [comment]
Fixed by #1289
